### ISSUE TITLE
fix(ally): unsemantic `button`

### ISSUE THREAD

**Issue by mfranzke** [calcom/cal.com#16587]
A close `button` is being implemented as `span` HTML element, and that for both not accessible for screenreader and keyboard users, as well as missing a textual description, currently only including a character.

https://github.com/calcom/cal.com/blob/7725adedc84c4afb319254a077094e2bd71c312c/packages/embeds/embed-core/src/ModalBox/ModalBoxHtml.ts#L44
